modules/docextract/lib/refextract_re.py

```diff
@@ -631,13 +631,13 @@ def compute_months():
 ## This pattern matches both url (http) and 'doi:' or 'DOI' formats
 re_doi = (re.compile(ur"""
     ((\(?[Dd][Oo][Ii](\s)*\)?:?(\s)*)       # 'doi:' or 'doi' or '(doi)' (upper or lower case)
-    |(https?://dx\.doi\.org\/))?            # or 'http://dx.doi.org/'    (neither has to be present)
-    (10\.                                   # 10.                        (mandatory for DOI's)
+    |(https?://(dx\.)?doi\.org\/))?         # or 'http://(dx.)doi.org/'  (neither has to be present)
+    (?P<doi>10\.                            # 10.                        (mandatory for DOI's)
     \d{4}                                   # [0-9] x4
-    /                                       # /
+    (/|%2f)                                 # / (possibly urlencoded)
     [\w\-_:;\(\)/\.<>]+                     # any character
     [\w\-_:;\(\)/<>])                       # any character excluding a full stop
-    """, re.VERBOSE))
+    """, re.VERBOSE + re.IGNORECASE))
```

Comment:
do not use `+` for flags, use `XOR`

`+` will work most of the time (since the flags are bitmasks in power of 2), but not when you accidentally use the same flag twice

```
In [11]: re.T, re.I, re.L, re.M, re.DOTALL, re.U, re.VERBOSE, re.DEBUG
Out[11]: (1, 2, 4, 8, 16, 32, 64, 128)

In [12]: re.T + re.I
Out[12]: 3

In [13]: re.T|re.I
Out[13]: 3

In [14]: re.T|re.I|re.L
Out[14]: 7

In [15]: re.T+re.I+re.L
Out[15]: 7

In [16]: re.I|re.I
Out[16]: 2

In [17]: re.I+re.I
Out[17]: 4
```
bazingaa!